scripts/eurostat/health_determinants/bmi/process.py

```diff
@@ -0,0 +1,555 @@
+                pvs.append(f"globalActivityLimitationindicator: dcs:{prop}")
+                name_pv.append(f"{prop}")
+            elif "weight" in prop \
+                or "Obese" in prop or "Obesity" in prop:
+                pvs.append(f"healthBehavior: dcs:{prop}")
+                name_pv.append(f"{prop}")
+        name_prop = ", ".join(name_pv)
+        name_prop = name_prop.replace(", Among", " Among")
+        name_prop = name_prop.rstrip(', ')
+        name_prop = name_prop.rstrip('with')
+        # Adding spaces before every capital letter,
+        # to make SV look more like a name.
+        name_prop = re.sub(r"(\w)([A-Z])", r"\1 \2", name_prop)
+        name_prop = "name: \"" + name_prop + "\""
+        name_prop = name_prop.replace("Percentage,", "Percentage")
+        name_prop = name_prop.replace("Population:,", "Population:")
+        name_prop = name_prop.replace("Among,", "Among")
+        name_prop = name_prop.replace("ACitizen", "A Citizen")
+        pvs.append(name_prop)
+        mcf_nodes.append(
+            _MCF_TEMPLATE.format(dcid=dcid, xtra_pvs='\n'.join(pvs)))
+    mcf = '\n'.join(mcf_nodes)
+    # Writing Genereated MCF to local path.
+    with open(mcf_file_path, 'w+', encoding='utf-8') as f_out:
+        f_out.write(mcf.rstrip('\n'))
+    # pylint: enable=too-many-statements
+
+
+def process(input_files: list, cleaned_csv_file_path: str, mcf_file_path: str,
+            tmcf_file_path: str):
+    """
+    Process Input Raw Files and apply transformations to generate final
+    CSV, MCF and TMCF files.
+    """
+
+    final_df = pd.DataFrame()
+    sv_list = []
+    f_names = []
+    for file_path in input_files:
+        file_name_with_ext = os.path.basename(file_path)
+        f_names.append(file_name_with_ext)
+        file_name_without_ext = os.path.splitext(file_name_with_ext)[0]
+        file_to_function_mapping = {
+            "hlth_ehis_bm1e": _age_sex_education,
+            "hlth_ehis_bm1i": _age_sex_income,
+            "hlth_ehis_bm1u": _age_sex_degree_urbanisation,
+            "hlth_ehis_bm1b": _age_sex_birth_country,
+            "hlth_ehis_bm1c": _age_sex_citizenship_country,
+            "hlth_ehis_bm1d": _age_sex_acitivity_limitation,
+            "hlth_ehis_de1": _age_sex_education_history,
+            "hlth_ehis_de2": _age_sex_income_history
+        }
+        df = pd.read_csv(file_path, sep='\t', header=0)
+        df = file_to_function_mapping[file_name_without_ext](df)
+        df['SV'] = df['SV'].str.replace('_Total', '')
+        final_df = pd.concat([final_df, df])
+        sv_list += df["SV"].to_list()
+
+    final_df.sort_values(by=['time', 'geo', 'SV', 'observation'], inplace=True)
+    final_df = final_df.drop_duplicates(subset=['time','geo','SV'],\
+        keep='first')
+    final_df['observation'] = final_df['observation'].astype(str)\
+        .str.strip()
+    # derived_df generated to get the year/SV/location sets
+    # where 'u' exist
```

Comment:
'u' denotes the measurement method 'EurostatRegionalStatistics_LowReliability' otherwise default measurement method is 'EurostatRegionalStatistics'. Some SV's of same geoId and different year has two measurement methods. So in this scenario, we are promoting measurement methods from 'EurostatRegionalStatistics' to 'EurostatRegionalStatistics_LowReliability'.

Reply:
please add a comment in the code explaining this